### ISSUE TITLE
Segfault from llvmjitcode test in Python 3.9

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#21935]
This test from the optional dependency tests fails with a segfault under Python 3.9:
```python
from sympy.printing.llvmjitcode import llvm_callable
from sympy import cse
from sympy.abc import a, b, n

e1 = a*a
e2 = a*a + b*b
e3 = cse([e1, e2])

f = llvm_callable([a, b], e3)
jit_res = f(0.1, 1.5)
```
That gives:
```console
$ python test.py 
Segmentation fault: 11
```
The test needs `llvmlite` to be installed (`pip install llvmlite`).

_Originally posted by @oscarbenjamin in https://github.com/sympy/sympy/issues/21885#issuecomment-903192133_

**Comment by oscarbenjamin** [comment]
CC @markdewing

**Comment by bjodah** [comment]
I think LLVM IR is a bit of a moving target, so I'm guessing llvmlite [might have gotten a bump](https://github.com/numba/llvmlite/blob/257f77f7417e5fbeea3aa3f7eaba176aec482fa9/CHANGE_LOG#L17)? Perhaps we need versioned printers (e.g. LLVM11JitPrinter, LLVM12JitPrinter, ...).

**Comment by oscarbenjamin** [comment]
Possibly fixed by #23513. See https://github.com/numba/llvmlite/issues/838
